staging/src/k8s.io/legacy-cloud-providers/aws/aws_loadbalancer.go

```diff
@@ -212,7 +222,26 @@ func (c *Cloud) ensureLoadBalancerv2(namespacedName types.NamespacedName, loadBa
 				frontendPort := mapping.FrontendPort
 				frontendProtocol := mapping.FrontendProtocol
 				nodePort := mapping.TrafficPort
-
+				healthCheckModify := func(mapping nlbPortMapping, tgrp *elbv2.TargetGroup) bool {
+					if !strings.EqualFold(mapping.HealthCheckProtocol, aws.StringValue(tgrp.HealthCheckProtocol)) ||
+						mapping.HealthCheckInterval != aws.Int64Value(tgrp.HealthCheckIntervalSeconds) {
+						return true
+					}
+					return false
+				}
+				if listener, ok := actual[frontendPort][frontendProtocol]; ok {
+					if targetGroup, ok := nodePortTargetGroup[nodePort]; ok {
+						if healthCheckModify(mapping, targetGroup) {
+							// If health check configuration changed, delete the existing listener so that a new
+							// listener and target group can get created with the new configuration
+							if err := c.deleteListenerV2(listener); err != nil {
```

Comment:
this looks hacky...since the listener don't need to be deleted. (only targetGroup need to be recreated)

Reply:
Most of the health check parameters are immutable, and not included in the target group naming. If any of the health check settings change, the new target group creation woudl fail (due to an existing one with the same name). Let me see what it would take to avoid deleting the listener.

Reply:
I was able to avoid the delete by modifying the target group naming logic.